src/packets.toit

```diff
@@ -11,6 +11,33 @@ import .topic_filter
 interface PacketIDAck:
   packet_id -> int
 
+class WillConfig:
+  retain/bool
+  qos/int
+  topic/string
+  payload/ByteArray
+
+
+  /**
+  Constructs the configuration of a last will message.
+
+  $topic
+    - the topic where the last will message will be published to
+
+  $payload
+    - content of the last will message
+
+  $qos
+    - quality of service level to be used
+
+  $retain
+    - if the message should be retained or not
+  */
+  constructor.from_byte_array .topic .payload --.qos --.retain=false:
```

Comment:
I would drop the name of this one (just use constructor), since the default for mqtt publish seems to be the byte array variant.